src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -45,146 +73,1195 @@ ActorInit En_Slime_InitVars = {
+}
+
+/**
+ * Update the actor's ice block timer to match state of ice block.
+ * When it reaches the "unset" value (255), return actor to idle.
+ */
+void EnSlime_IceBlockThaw(EnSlime* this, PlayState* play) {
+    s32 iceBlockTimerAdj;
+
+    this->actor.colorFilterTimer = 10;
+    if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+        iceBlockTimerAdj = (0.1f - this->actor.child->scale.y) * 10.0f * 255.0f;
+        this->iceBlockTimer = CLAMP(iceBlockTimerAdj, 0, 255);
+    } else {
+        this->actor.child = NULL;
+        iceBlockTimerAdj = this->iceBlockTimer + 10;
+        this->iceBlockTimer = CLAMP_MAX(iceBlockTimerAdj, 255);
+    }
+
+    if (this->iceBlockTimer == 255) {
+        this->collider.base.acFlags |= AC_ON;
+        this->actor.flags |= ACTOR_FLAG_1;
+        this->actor.flags &= ~ACTOR_FLAG_10;
+        EnSlime_SetupIdle(this);
+    }
+}
+
+/**
+ * Set speed to zero, and make undrawable.
+ * Initialize respawn timer.
+ */
+void EnSlime_SetupWaitForRevive(EnSlime* this) {
+    this->actor.draw = NULL;
+    this->actor.flags |= ACTOR_FLAG_10;
+    this->drawDmgEffAlpha = 0.0f;
+    this->actor.gravity = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.speed = 0.0f;
+    this->collider.base.ocFlags1 &= ~OC1_ON;
+    this->timer = this->respawnTime;
+    this->actionFunc = EnSlime_WaitForRevive;
+}
+
+/**
+ * Decrement the revival timer, or if it is zero,
+ * initialize revive.
+ */
+void EnSlime_WaitForRevive(EnSlime* this, PlayState* play) {
+    if (this->timer == 0) {
+        EnSlime_SetupRevive(this);
+    } else {
+        this->timer--;
+    }
+}
+
+/**
+ * Re-enable drawing, reset the actor's health and home,
+ * and setup and initialize respawn action.
+ */
+void EnSlime_SetupRevive(EnSlime* this) {
+    this->actor.draw = EnSlime_Draw;
+    this->actor.colChkInfo.health = sColChkInfoInit.health;
+    this->iceBlockTimer = 255;
+    Math_Vec3f_Copy(&this->actor.world.pos, &this->actor.home.pos);
+    this->timer = 0;
+    this->actor.colorFilterTimer = 0;
+    this->actor.home.rot.y = this->actor.yawTowardsPlayer;
+    Actor_PlaySfx(&this->actor, NA_SE_EN_SLIME_SURFACE);
+    this->actor.gravity = -2.0f;
+    this->actionFunc = EnSlime_Revive;
+}
+
+/**
+ * Update bubbling back up effect. After 21 frames, OC1_ON is set.
+ * After 28 frames, AC_ON is set and actor returns to idle movement.
+ */
+void EnSlime_Revive(EnSlime* this, PlayState* play) {
+    f32 rescaleFactor1;
+    f32 rescaleFactor2;
+
+    this->timer++;
+    if (this->timer == 28) {
+        this->actor.flags &= ~ACTOR_FLAG_10;
+        this->actor.flags |= ACTOR_FLAG_1;
+        this->collider.base.acFlags |= AC_ON;
+        this->actor.shape.rot.y = this->actor.home.rot.y;
+        EnSlime_SetupMoveInDirection(this);
+    } else {
+        if (this->timer < 12) {
+            rescaleFactor1 = this->timer * 0.0008333333f; // This is about 1/1200, but (1.0f/1200.0f) does not match
```

Comment:
Does `(1.0f / 12.0f) / 100.0f` work? I'm just taking a complete shot in the dark, so if it doesn't, no worries.